logger.go

```diff
@@ -121,7 +142,10 @@ func WithLogger(ctx context.Context, logger *Logger) context.Context {
 
 func FromContext(ctx context.Context) *Logger {
 	if logger, ok := ctx.Value(loggerKey{}).(*Logger); ok {
+		logger.ctx = ctx
 		return logger
 	}
-	return DefaultLogger()
+	l := DefaultLogger()
+	l.ctx = ctx
```

Comment:
This is problematic because it might end up mutating external state 😞 

e.g.

``` go
log := slog.New(slogctx.NewHandler(slog.NewTextHandler(os.Stdout, nil)))
for _, v := range []string{"foo", "bar"} {
  go func(v string) {
    ctx := slogctx.WithValues(ctx, "key", v)
    // this races because both goroutines will try to overwrite ctx on the same logger
    log := slogctx.FromContext(ctx).Info("hello!")
  }
}
```

we'll need to create a separate copy of the logger, a new type, or keep the Context funcs.

Reply:
Good catch!

WDYT about copying the logger?